src/main/java/com/linecorp/armeria/server/thrift/THttpService.java

```diff
@@ -500,7 +500,7 @@ private void decodeAndInvoke(
                 final TApplicationException cause = new TApplicationException(
                         TApplicationException.UNKNOWN_METHOD, "unknown method: " + header.name);
 
-                respond(serializationFormat, seqId, methodName, cause, res);
+                respond(ctx, serializationFormat, seqId, methodName, cause, res);
```

Comment:
Couldn't do that because it's the behavior of the official TMultiplexedProtocol implementation.